app/models/manageiq/providers/redhat/infra_manager/vm/operations.rb

```diff
@@ -8,4 +8,8 @@ module ManageIQ::Providers::Redhat::InfraManager::Vm::Operations
   def raw_destroy
     with_provider_object(&:destroy)
   end
+
+  def raw_unregister
+    with_provider_object(:version => 4, &:unregister)
```

Comment:
@masayag what will happen if this is not using the v4 api?

Reply:
@agrare The request will fail with the following error:

[----] E, [2018-06-11T00:18:27.589957 #11273:11190f8] ERROR -- : MIQ(MiqQueue#m_callback) Message id: [12982]: version 4 of the api is not supported by the provider

Reply:
Hm okay we should probably have a supports feature mixin for unregister so that we can say unsupported if version 3.

@masayag can you put in a PR to add that to master?

Reply:
@agrare this action is being called directly from the automation script and not from the UI, Meaning there is no mention of that action in https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/helpers/application_helper/toolbar/x_vm_center.rb
Therefore adding it to the supported_feature_mixin will not assist in validating if it is supported or not in this use case.

I can add a specific validation to be used within this method, unless there is a better way of doing so.

Reply:
@masayag I still hold out hope that automate will also check for supported features before calling methods, and we might want to add unregister to the UI at some point.